core/src/worker/workflow/machines/child_workflow_state_machine.rs

```diff
@@ -79,6 +82,15 @@ fsm! {
     Cancelled --(Cancel) --> Cancelled;
     Cancelled --(ChildWorkflowExecutionCancelled,
         on_child_workflow_execution_cancelled) --> Cancelled;
+    // Completions of any kind after cancellation are acceptable for abandoned children
+    Cancelled --(ChildWorkflowExecutionCompleted(Option<Payloads>),
```

Comment:
Wonder if one day you may want to expand this macro format to build in "guards" so you can just add an if-abandoned conditional

Reply:
Maybe. I don't want to make the DSL to complex when you can handle that in the handler easily.